st2common/st2common/validators/api/reactor.py

```diff
@@ -65,11 +66,14 @@ def validate_trigger_parameters(trigger_type_ref, parameters):
     if not trigger_type_ref:
         return None
 
-    if trigger_type_ref not in SYSTEM_TRIGGER_TYPES:
-        # Not a system trigger, skip validation for now
+    trigger_type = triggers.get_trigger_type_db(trigger_type_ref)
+    if trigger_type_ref in SYSTEM_TRIGGER_TYPES:
+        parameters_schema = SYSTEM_TRIGGER_TYPES[trigger_type_ref]['parameters_schema']
```

Comment:
After some more thought, I think we should have a single config option.

Perhaps, something along the lines of:

```ini
[system]
validate_trigger_parameters = False
```

The description should also be explicit and say something along those lines of "True to enable trigger parameter validation during rule creation and when dispatching triggers inside a sensor".

For backward compatibility, we should default it to false like you did with the config option you added.

We also need to document this new flag and behavior in the upgrade notes section of document (clarify how it works and affects the system).